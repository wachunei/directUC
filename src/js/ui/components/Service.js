import styled from "styled-components";

const Service = styled.div`
  background-color: ${({ theme }) => theme.colors.serviceBackground};
  &:not(:last-child) {
    border-bottom: 1px solid ${(props) => props.theme.horizontalSeparator};
    padding-bottom: 20px;
  }
  margin-bottom: 20px;
`;

export default Service;
