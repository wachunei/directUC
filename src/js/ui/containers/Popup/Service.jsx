import styled from "styled-components";
import Button from "../../components/Button";

const ServiceContainer = styled.div`
  &:not(:last-child) {
    margin-bottom: 0.5rem;
  }
`;

function Service(props) {
  return (
    <ServiceContainer>
      <Button block primary {...props} />
    </ServiceContainer>
  );
}

export default Service;
