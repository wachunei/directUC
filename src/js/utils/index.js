export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
