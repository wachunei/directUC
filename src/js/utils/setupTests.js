// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
