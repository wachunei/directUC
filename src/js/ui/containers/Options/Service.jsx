import React from "react";
import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import Service from "../../components/Service";
import SubTitle from "../../components/SubTitle";
import FormControl from "../../components/FormControl";
import Checkbox from "../../components/Checkbox";

import ServiceOption from "./ServiceOption";

function ServiceContainer({
  loggedIn,
  service,
  serviceKey,
  userOptions,
  disabledToggleDisplay,
}) {
  const dispatch = useDispatch();
  const { options, name } = service;

  const handleOptionChange = async (e) => {
    const {
      target: { name: inputName, value, type, checked },
    } = e;
    await dispatch({
      type: `servicesActions.${serviceKey}.setOption`,
      payload: {
        option: inputName,
        value: type === "checkbox" ? checked : value,
      },
    });
  };

  return (
    <>
      <Service>
        <SubTitle>{name}</SubTitle>
        <FormControl>
          <Checkbox
            name="display"
            disabled={
              !loggedIn || (userOptions.display && disabledToggleDisplay)
            }
            checked={userOptions.display}
            onChange={handleOptionChange}
          >
            Mostrar Bot√≥n
          </Checkbox>
        </FormControl>

        {options &&
          Object.entries(options).map(([optionKey, option]) => {
            if (optionKey === "display") {
              return null;
            }

            return (
              <ServiceOption
                key={optionKey}
                name={optionKey}
                option={option}
                userOptions={userOptions}
                value={userOptions[optionKey]}
                disabled={!loggedIn}
                onChange={handleOptionChange}
              />
            );
          })}
      </Service>
    </>
  );
}

ServiceContainer.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  service: PropTypes.shape({
    name: PropTypes.string,
    options: PropTypes.object,
  }).isRequired,
  serviceKey: PropTypes.string.isRequired,
  userOptions: PropTypes.shape({
    display: PropTypes.bool,
  }).isRequired,
  disabledToggleDisplay: PropTypes.bool.isRequired,
};

export default ServiceContainer;
