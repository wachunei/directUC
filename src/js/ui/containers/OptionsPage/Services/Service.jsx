import React, {
  forwardRef,
  useCallback,
  useImperativeHandle,
  useRef,
  useState,
} from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { useDispatch } from "react-redux";
import Service from "../../../components/Service";
import SubTitle from "../../../components/SubTitle";
import FormControl from "../../../components/FormControl";
import Checkbox from "../../../components/Checkbox";
import Box from "../../../components/Box";

import ServiceOption from "./ServiceOption";

const DragHandle = styled.span`
  opacity: 0;
  transition: opacity 0.2s;
`;

const StyledService = styled(Service)`
  &:hover {
    ${DragHandle} {
      opacity: ${({ loggedIn }) => (loggedIn ? 1 : 0)};
    }
  }
`;

const ServiceContainer = forwardRef(function ServiceContainerComponent(
  {
    loggedIn,
    service,
    serviceKey,
    userOptions,
    disabledToggleDisplay,
    onDragStart,
    onDragEnd,
  },
  ref
) {
  const serviceRef = useRef(null);
  useImperativeHandle(ref, () => ({
    key: serviceKey,
    current: serviceRef.current,
  }));
  const dispatch = useDispatch();
  const { options, name } = service;
  const [draggable, setDraggable] = useState(false);
  const [dragging, setDragging] = useState(false);

  const handleOptionChange = async (e) => {
    const {
      target: { name: inputName, value, type, checked },
    } = e;
    await dispatch({
      type: `servicesActions.${serviceKey}.setOption`,
      payload: {
        option: inputName,
        value: type === "checkbox" ? checked : value,
      },
    });
  };

  const handleDragHandlerEnter = useCallback(() => setDraggable(loggedIn), [
    loggedIn,
  ]);
  const handleDragHandlerLeave = useCallback(() => setDraggable(false), []);
  const handleDragStart = useCallback(() => {
    setDragging(true);
    onDragStart(serviceKey);
  }, []);

  const handleDragEnd = useCallback(() => {
    setDragging(false);
    onDragEnd();
  }, []);

  return (
    <StyledService
      draggable={draggable}
      onDragStart={handleDragStart}
      onDragEnd={handleDragEnd}
      dragging={dragging}
      loggedIn={loggedIn}
      ref={serviceRef}
    >
      <Box
        horizontal
        between
        middle
        style={loggedIn ? { cursor: dragging ? "grabbing" : "grab" } : null}
        onMouseEnter={handleDragHandlerEnter}
        onMouseLeave={handleDragHandlerLeave}
      >
        <SubTitle>{name}</SubTitle>
        <DragHandle>â ¿</DragHandle>
      </Box>

      <FormControl>
        <Checkbox
          name="display"
          disabled={!loggedIn || (userOptions.display && disabledToggleDisplay)}
          checked={userOptions.display}
          onChange={handleOptionChange}
        >
          Activar
        </Checkbox>
      </FormControl>

      {options &&
        Object.entries(options).map(([optionKey, option]) => {
          if (optionKey === "display") {
            return null;
          }

          return (
            <ServiceOption
              key={optionKey}
              name={optionKey}
              option={option}
              userOptions={userOptions}
              value={userOptions[optionKey]}
              disabled={!loggedIn}
              onChange={handleOptionChange}
            />
          );
        })}
    </StyledService>
  );
});

ServiceContainer.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  service: PropTypes.shape({
    name: PropTypes.string,
    options: PropTypes.object,
  }).isRequired,
  serviceKey: PropTypes.string.isRequired,
  userOptions: PropTypes.shape({
    display: PropTypes.bool,
  }).isRequired,
  disabledToggleDisplay: PropTypes.bool.isRequired,
  onDragStart: PropTypes.func.isRequired,
  onDragEnd: PropTypes.func.isRequired,
};

export default ServiceContainer;
