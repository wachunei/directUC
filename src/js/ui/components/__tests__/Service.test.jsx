import Service from "../Service";
import { render, getElement } from "./utils";

describe("Service Component", () => {
  test("should match snapshot", () => {
    const { container } = render(<Service />);
    expect(getElement(container)).toMatchSnapshot();
  });
  test("should match snapshot with lightTheme", () => {
    const { container } = render(<Service />, "lightTheme");
    expect(getElement(container)).toMatchSnapshot();
  });
  test("should match snapshot with darkTheme", () => {
    const { container } = render(<Service />, "darkTheme");
    expect(getElement(container)).toMatchSnapshot();
  });
});
