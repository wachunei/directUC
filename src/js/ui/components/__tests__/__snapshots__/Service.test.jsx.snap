// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service Component should match snapshot 1`] = `
.c0 {
  background-color: #FAFAFA;
  margin-bottom: 20px;
  opacity: 1;
}

.c0:not(:last-child) {
  border-bottom: 1px solid #E0E0E0;
  padding-bottom: 20px;
}

<div
  class="c0"
/>
`;

exports[`Service Component should match snapshot with darkTheme 1`] = `
.c0 {
  background-color: #202020;
  margin-bottom: 20px;
  opacity: 1;
}

.c0:not(:last-child) {
  border-bottom: 1px solid #303030;
  padding-bottom: 20px;
}

<div
  class="c0"
/>
`;

exports[`Service Component should match snapshot with lightTheme 1`] = `
.c0 {
  background-color: #FAFAFA;
  margin-bottom: 20px;
  opacity: 1;
}

.c0:not(:last-child) {
  border-bottom: 1px solid #E0E0E0;
  padding-bottom: 20px;
}

<div
  class="c0"
/>
`;
